server/pulp/plugins/rsync/publish.py

```diff
@@ -0,0 +1,510 @@
+        auth_type = self.get_config().flatten()["remote"]['auth_type']
+
+        # -e 'ssh -l l_name -i key_path'
+        # use shared ssh connection for other threads
+        cmd = ['ssh', '-l', user]
+        if auth_type == "publickey":
+            key = self.get_config().flatten()["remote"]['key_path']
+            cmd += ['-i', key]
+        cmd += ['-o', 'StrictHostKeyChecking no',
+                '-o', 'UserKnownHostsFile /dev/null',
+                '-S', '/tmp/rsync_distributor-%r@%h:%p',
+                '-o', 'ControlMaster auto',
+                '-o', 'ControlPersist 10']
+        if self.get_config().flatten()["remote"]['auth_type'] == 'password':
+            cmd = ['sshpass', '-p', self.get_config().flatten()['remote']['ssh_password']] + cmd
+        if args:
+            cmd += args
+        return cmd
+
+    def make_authentication(self):
+        """
+        Returns a string representation of command for authenticating against a remote server.
+
+        :return: str. e.g., '-e ssh -l ssh_login_user -i /key_path'
+        :rtype: str
+        """
+        if self.get_config().flatten()["remote"]['auth_type'] == 'local':
+            return ()
+        ssh_parts = []
+        for arg in self.make_ssh_cmd():
+            if " " in arg:
+                ssh_parts.append('"%s"' % arg)
+            else:
+                ssh_parts.append(arg)
+
+        return ['-e', " ".join(ssh_parts)]
+
+    def make_full_path(self, relative_path):
+        """
+        Returns full path of a relative path on the remote server.
+
+        :param relative_path: relative path
+        :type relative_path: str
+        """
+        return os.path.join(self.get_config().flatten()["remote"]['root'], relative_path)
+
+    def make_destination(self, relative_path):
+        """
+        Parse from self.config information to make up a hostname and remote path used for rsync
+        command.
+
+        :return: str of the combination of user, host, and dir.
+        :rtype:  str
+        """
+        if self.get_config().flatten()["remote"]['auth_type'] == 'local':
+            return self.get_config().flatten()["remote"]['root']
+        relative_path = relative_path.lstrip("/")
+
+        user = self.get_config().flatten()["remote"]['ssh_login']
+        host = self.get_config().flatten()["remote"]['host']
+        remote_root = self.get_config().flatten()["remote"]['root']
+        return '%s@%s:%s' % (user, host, os.path.join(remote_root, relative_path))
+
+    def call(self, args, include_args_in_output=True):
+        """
+        A wrapper around kobo.shortcuts.run. If ssh_exchange_identification or
+        max-concurrent-connections exceptions are thrown by ssh, up to 10 retries follows.
+        """
+        for t in xrange(10):
+            rv, out = kobo.shortcuts.run(cmd=args, can_fail=True)
+            possible_known_exceptions = \
+                ("ssh_exchange_identification:" or "max-concurrent-connections=25") in out
```

Comment:
I don't think this works how you're hoping. I expect the expression in parenthesis will be evaluated to just the first string, and the second string will not be searched for in ``out``. To illustrate the problem:

```
>>> (4 or 6) in range(5)
True
>>> (6 or 4) in range(5)
False
```